networking/src/ModelHelper.ts

```diff
@@ -100,20 +100,18 @@ export abstract class ModelHelper {
         }
 
         const correctAnswer = this.getCorrectAnswer(question)
+        const currentQuestion = gameSession?.questions[gameSession?.currentQuestionIndex ?? 0]
+        let originalScore = gameSession?.teams?.find(teamElement => teamElement.id === team.id)?.score ?? 0 
 
-        return answers!.reduce((score: number, answer: ITeamAnswer | null) => {
-            if (isNullOrUndefined(answer)) {
-                return score
-            }
-
-            if (answer.isTrickAnswer) {
-                return score + this.calculateBasicModeWrongAnswerScore(gameSession, answer.text, question.id)
-            } else {
-                console.log(`${answer.text} === ${correctAnswer?.text} = ${answer.text === correctAnswer?.text}`)
-                return score + (
-                    answer.text === correctAnswer?.text ? this.correctAnswerScore : 0)
-            }
-        }, 0)
+        if (answers.some(answer => answer?.isTrickAnswer===true)){
+          let newScore = originalScore += ModelHelper.calculateBasicModeWrongAnswerScore(gameSession, answers?.find(answer => answer?.isTrickAnswer==true)?.text ?? '', currentQuestion.id)
+          return newScore
+        }
+        else if (answers.some(answer => answer?.isChosen===true && answer?.text == correctAnswer?.text)){
+          let newScore = originalScore += this.correctAnswerScore
+          return newScore
+        }
+        return originalScore
     }
```

Comment:
I rewrote this to use team.score to calculate the ongoing score. I think this might be smoother than having to recalculate the score of all questions using a .reduce

Reply:
Can you also write some tests for these changes and also make sure the existing tests pass?

Reply:
This was a great comment. Looking into the tests exposed some errors in the logic that I hadn't caught. I've updated the tests to reflect single question and multiple question cases and they both pass.